SAP/luigi
# Fix context update bug

**Description**
Navigating inside of a micro-frontend (ex: Angular based) using its internal angular routing is facilitated using **withoutSync**. However, even though the navigation is successfully prevented with this function, there seems to be no context update upon internal navigation. 

**Changes**
- Have the Core [send a message to the iframe ](https://github.com/SAP/luigi/pull/1953/files#diff-33caff53e1abed9c033da822ec4bb4c2eee2ea754e6760968883ccf731bb8868R341-R348) when withoutSync is activated. An additional **withoutSync** parameter is added. The rest of the message body is the same.
- [Catch the event message on the Client side](https://github.com/SAP/luigi/pull/1953/files#diff-ea2419053062f053bee8d5e4a11c1df366970c29d50601d12ac2fe1ceebcb149R107-R117) in the lifecycle manager and handle it differently if withoutSync is activated:
    -  Omit navigation if withoutSync is present
    -  Add extra parameters to the context as useful information for the virtualTree structure

**How to test**
1. Open **e2e-test-application** and navigate to **projects/pr2/developers**
2. Click the withoutSync link and check the console for Context Changed print outs. 
3. The printed context should have **pathParams** and **viewUrl** as properties. 
 
**Related issue(s)**
Fixes #1940
